pkg/storegateway/bucket.go

```diff
@@ -1598,24 +1574,6 @@ func (s *bucketBlockSet) getFor(mint, maxt, maxResolutionMillis int64, blockMatc
 	return bs
 }
 
-// labelMatchers verifies whether the block set matches the given matchers and returns a new
-// set of matchers that is equivalent when querying data within the block.
-func (s *bucketBlockSet) labelMatchers(matchers ...*labels.Matcher) ([]*labels.Matcher, bool) {
```

Comment:
_Note to reviewers._

I think it's safe removing it. Why? My reasoning is:
- This function takes in input the label matchers from the query, and return a set of label matchers which excludes external labels of the block set (because external labels are expected to be set on the block's metadata and not on the series inside the block).
- However, when we create `bucketBlockSet` in Mimir, the `bucketBlockSet.labels` is always empty when using the bucket index, because it's not among the set of data included in the bucket index: https://github.com/grafana/mimir/blob/e9bba6b079e5d6e71037b17aab74907391975b36/pkg/storage/tsdb/bucketindex/index.go#L110-L123
- Pratically, `bucketBlockSet.labels` is always empty in Mimir, so we already have only 1 `bucketBlockSet` with empty external labels